### PR TITLE
Create Group and Add Member functionality

### PR DESCRIPTION
**Pull request by pattmax00**

### CODE REVIEW

**Review by xeelo2000 — DISMISSED** [review]

File: AcademicReward/Database/GroupDatabase.cs [hunk 1]
```diff
@@ -0,0 +1,134 @@
+﻿using AcademicReward.Resources;
+using Npgsql;
+using System.Collections.ObjectModel;
+
+namespace AcademicReward.Database
+{
```
Comment (Wil-LaLonde) [inline AcademicReward/Database/GroupDatabase.cs:6]: Can we update the comments in this class? Some are referring to gathering tasks even though this is for group stuff?

File: AcademicReward/Database/GroupProfileRelationship.cs [hunk 2]
```diff
@@ -50,5 +50,71 @@ public static ObservableCollection<Profile> getProfilesInGroup(Group group)
...
+            {
+                Console.WriteLine("Error while adding profile to group: {0}", e);
+            }
+        }
+
+        public static Profile findByUsername(string username)
+        {
+            try
+            {
+                using var con = new NpgsqlConnection(InitializeConnectionString());
```
Comment (Wil-LaLonde) [inline AcademicReward/Database/GroupProfileRelationship.cs:81]: I guess I'm just curious, what happens if the username is not found? Or what if the username being added is an admin? What are we reporting back to the user?

File: AcademicReward/Logic/GroupLogic.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    /// Reviewer: 
+    /// </summary>
+    public class GroupLogic : ILogic
+    {
+        private IDatabase GroupDB;
+
+        /// <summary>
+        /// LoginGroupLogic constructor
+        /// </summary>
+        public GroupLogic()
```
Comment (Wil-LaLonde) [inline AcademicReward/Logic/GroupLogic.cs:19]: Please update comments, also, are we going to show anything to the user if an error occurs? We really shouldn't always be returning that no error occurred.

File: AcademicReward/PopUps/AddMemberPopUp.xaml.cs [hunk 4]
```diff
@@ -8,11 +13,34 @@ namespace AcademicReward.PopUps;
...
+		// Grab the text from the entry box
+		String memberName = MemberNameEntry.Text;
+
+		// Add member logic
+		var addMemberLogic = new AddMemberLogic();
+
+		// Create an array with two elements, the current group and the member name
+		object[] obj = new object[] { memberName, group };
+
+		var error = addMemberLogic.AddItemWithArgs(obj);
```
Comment (Wil-LaLonde) [inline AcademicReward/PopUps/AddMemberPopUp.xaml.cs:40]: Are you doing anything with this error var?

File: AcademicReward/Views/CreateGroupPage.xaml.cs [hunk 5]
```diff
@@ -7,8 +10,32 @@ namespace AcademicReward.Views;
...
+        // Instantiate groupLogic
+        groupLogic = new GroupLogic();
+        InitializeComponent();
+    }
+
+    private void CreateGroupButtonClicked(object sender, EventArgs e)
+    {
+        // Grab the text from the entry boxes
+        var groupName = GroupNameEntry.Text;
+        var groupDescription = GroupDescriptionEntry.Text;
```
Comment (Wil-LaLonde) [inline AcademicReward/Views/CreateGroupPage.xaml.cs:26]: What about if the group name is empty? Or if it exceeds the database limits?

**Review by Wil-LaLonde — APPROVED** [review]
Looks good, thank you for making those changes

**Pull request merged**